packages/taco/test/conditions/context.test.ts

```diff
@@ -242,3 +243,36 @@ describe('context', () => {
     });
   });
 });
+
+describe('param or context param schema', () => {
+  it('accepts a plain string', () => {
+    expect(ParamOrContextParamSchema.safeParse('hello').success).toBe(true);
+  });
+
+  it('accepts a context param', () => {
+    expect(ParamOrContextParamSchema.safeParse(':hello').success).toBe(true);
+  });
+
+  it('accepts a number', () => {
+    expect(ParamOrContextParamSchema.safeParse(123).success).toBe(true);
+  });
+
+  it('accepts a boolean', () => {
+    expect(ParamOrContextParamSchema.safeParse(true).success).toBe(true);
+  });
+
+  it('accepts an array', () => {
+    expect(
+      ParamOrContextParamSchema.safeParse([1, 'hello', true]).success,
+    ).toBe(true);
+  });
```

Comment:
Nice job on the tests. If we are going to be this prescriptive we need to be really comprehensive with our testing, and add as many cases as we can think of in terms of:
- parameters to function calls (think of passing structs etc.)
- returns from function calls since this schema is also used for the returnValueTest `value`.

Can you also add tests to try the following:

```typescript
  it('accepts nested arrays', () => {
    expect(
      ParamOrContextParamSchema.safeParse([1, "hello", [123, 456, "hello", true]]).success,
    ).toBe(true);
  });
  ```
  and this:
  
  ```typescript
  it('accepts nested arrays', () => {
    expect(
      ParamOrContextParamSchema.safeParse([1, "hello", [123, ":hello"]]).success,
    ).toBe(true);
  });
  ```

Do they pass? If they don't pass, and we go down this route - I think you may need to do a repetition of ParamSchema on itself, `lazy` I think zod called it - similar to what was done for CompoundCondition.

Reply:
- Floats are already covered by `z.number()`. One can use `z.number().int()` to further discriminate number types, etc. 
- Bytes as hex and string addresses should work just fine with `z.string()`.
- I added some tests to cover nested structures. On that note, I want to write more tests before release

Reply:
> - Floats are already covered by z.number(). One can use z.number().int() to further discriminate number types, etc.
> - Bytes as hex and string addresses should work just fine with z.string().

Indeed. I just meant that it would be good to have explicit tests for those anyway to prevent future regressions and clarify expectations.